gateway-ha/src/main/java/io/trino/gateway/ha/handler/QueryIdCachingProxyHandler.java

```diff
@@ -264,42 +259,29 @@ public boolean handleAuthRequest(HttpServletRequest request)
     @Override
     public String rewriteTarget(HttpServletRequest request)
     {
-        /* Here comes the load balancer / gateway */
-        String backendAddress = "http://localhost:" + serverApplicationPort;
-
-        // Only load balance trino query APIs.
-        if (isPathWhiteListed(request.getRequestURI())) {
-            String queryId = extractQueryIdIfPresent(request);
+        if (!isPathWhiteListed(request.getRequestURI())) {
+            return buildUriWithNewBackend(applicationEndpoint, request);
+        }
 
-            // Find query id and get url from cache
-            if (!Strings.isNullOrEmpty(queryId)) {
-                backendAddress = routingManager.findBackendForQueryId(queryId);
-            }
-            else {
-                String routingGroup = routingGroupSelector.findRoutingGroup(request);
-                String user = request.getHeader(USER_HEADER);
-                if (!Strings.isNullOrEmpty(routingGroup)) {
-                    // This falls back on adhoc backend if there are no cluster found for the routing group.
-                    backendAddress = routingManager.provideBackendForRoutingGroup(routingGroup, user);
-                }
-                else {
-                    backendAddress = routingManager.provideAdhocBackend(user);
-                }
-            }
-            // set target backend so that we could save queryId to backend mapping later.
-            ((MultiReadHttpServletRequest) request).addHeader(PROXY_TARGET_HEADER, backendAddress);
+        if (isUseStickyRouting(request)) {
+            String backend = getPreviousBackend(request);
+            logRewrite(backend, request);
+            return buildUriWithNewBackend(backend, request);
         }
-        if (isAuthEnabled() && request.getHeader("Authorization") != null) {
-            if (!handleAuthRequest(request)) {
-                // This implies the AuthRequest was not authenticated, hence we error out from here.
-                log.info("Could not authenticate Request: " + request);
-                return null;
-            }
+
+        String backend = getBackendFromRoutingGroup(request);
+        // set target backend so that we could save queryId to backend mapping later.
+        ((MultiReadHttpServletRequest) request).addHeader(PROXY_TARGET_HEADER, backend);
+        logRewrite(backend, request);
+        return buildUriWithNewBackend(backend, request);
+    }
+
+    private void logRewrite(String newBackend, HttpServletRequest request)
+    {
+        if (!log.isInfoEnabled()) {
+            return;
```

Comment:
nit: I would remove this block. We basically don't use log level to change the method behavior in Trino.